python/cudf/cudf/tests/test_indexing.py

```diff
@@ -123,6 +127,51 @@ def test_series_indexing(i1, i2, i3):
             assert series[i] == a1[i]
 
 
+@pytest.mark.parametrize(
+    "arg",
+    [
+        1,
+        -1,
+        "b",
+        np.int32(1),
+        np.uint32(1),
+        np.int8(1),
+        np.uint8(1),
+        np.int16(1),
+        np.uint16(1),
+        np.int64(1),
+        np.uint64(1),
+    ],
+)
+def test_series_get_item_iloc_defer(arg):
+    # Indexing for non-numeric dtype Index
+    ps = pd.Series([1, 2, 3], index=pd.Index(["a", "b", "c"]))
+    gs = cudf.from_pandas(ps)
+
+    expect = ps[arg]
+    got = gs[arg]
+
+    assert_eq(expect, got)
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    ["int32", "int16", "int8", "int64", "uint32", "uint16", "uint8", "uint64"],
+)
+def test_series_iloc_defer_cudf_Scalar(dtype):
+    ps = pd.Series([1, 2, 3], index=pd.Index(["a", "b", "c"]))
+    gs = cudf.from_pandas(ps)
+
+    arg = cudf.Scalar(1, dtype=dtype)
+    got = gs[arg]
+
+    with pytest.raises(
+        TypeError, match=re.escape(f"'{arg}' is an invalid key"),
+    ):
+        expect = ps[arg]
+        assert_eq(expect, got)
```

Comment:
We shouldn't need to test if Pandas support `cudf.Scalar`. Instead, just do `expected = 2` and `assert_eq(expected, got)`